msal/application.py

```diff
@@ -271,7 +271,7 @@ def _get_authority_aliases(self, instance):
         if not self.authority_groups:
             resp = requests.get(
                 "https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https://login.microsoftonline.com/common/oauth2/authorize",
-                headers={'Accept': 'application/json'})
+                headers={'Accept': 'application/json'}, verify=self.verify)
```

Comment:
Thanks for the catch! And, since we are modifying this anyway, I would suggest to also wire up the `proxies` and `timeout`.

                headers={'Accept': 'application/json'},
                verify=self.verify, proxies=self.proxies, timeout=self.timeout)